screens/search_screen/search.tsx

```diff
@@ -40,12 +43,25 @@ const Search: FC<SearchScreenProps> = (props) => {
 
             let currentLocation: any = await Location.getCurrentPositionAsync({});
             setLocation(currentLocation);
+
           })();
         }, []);
       }
 
-      let currentLocation: any = await Location.getCurrentPositionAsync({});
-      setLocation(currentLocation);
+      let currentLocation = await Location.getCurrentPositionAsync({});
+
+      let latitude = currentLocation.coords.latitude;
+      let longitude = currentLocation.coords.longitude;
+
+      let userLocation = latitude + ";" + longitude;
+
+      setLocation(userLocation);
+
+      dispatch(getLocation(location))
+      dispatch(getIsGeoLoading(true))
+
+      props.navigation.push("Main");
```

Comment:
Do we want to navigate if the permission request failed?

Reply:
No, we shouldn't let this happen. Should all this logic be moved into the if statement maybe?

Reply:
As far as I can see, this will immediately navigate when the component renders.

Reply:
Ah, I see what you mean.